File: frontend/src/typings/RootFolder.ts
import ModelBase from 'App/ModelBase';

export interface UnmappedFolder {
  name: string;
  path: string;
  relativePath: string;
}

interface RootFolder extends ModelBase {
  id: number;
  path: string;
  accessible: boolean;
  freeSpace?: number;
  unmappedFolders: UnmappedFolder[];
}

export default RootFolder;

File: src/Workarr/RootFolders/RootFolder.cs
using Workarr.Datastore;

namespace Workarr.RootFolders
{
    public class RootFolder : ModelBase
    {
        public string Path { get; set; }
        public bool Accessible { get; set; }
        public long? FreeSpace { get; set; }
        public long? TotalSpace { get; set; }

        public List<UnmappedFolder> UnmappedFolders { get; set; }
    }
}
